File: SY/Core_Subjects/Operating_Systems/C++/critical_section.cpp
#include<iostream>
#include<thread>
using namespace std;
int count;
void inr(){
	for(int i = 0; i < 100000; ++i){
		++count;
	}
}
int main(){
	thread t1(inr);
	thread t2(inr);

	cout << "Before : " << count << "\n";

	t1.join();
	t2.join();
	cout << "After : " << count << '\n';
	return 0;
}

File: SY/Core_Subjects/Operating_Systems/Python/critical_section.py
from threading import *

lock = Lock()

x = 0

def inr():
    for i in range(1000000):
        global x
        x += 1

if __name__ == '__main__':
    t1 = Thread(target = inr)
    t2 = Thread(target = inr)

    t1.start()
    t2.start()

    t1.join()
    t2.join()

    print(x)
